src/main/java/com/petros/bibernate/dao/EntityPersister.java

```diff
@@ -58,9 +55,9 @@ public EntityPersister(DataSource dataSource, boolean showSql) {
      * @param <T>         the type of the entity
      * @return the entity with the specified identifier, or null if not found
      */
-    public <T> T findById(Class<T> entityClass, Object idValue) {
+    public <T> T findById(Class<T> entityClass, Object idValue, Connection connection) {
```

Comment:
Current implementation allows to open connection on execution method flush. You propose to create connection on session creation stage. I'm not sure that it would be better solution. EntityPersister in Hibernate has similar implementation, but instead of Connection, session instance is passed to all methods.
<img width="821" alt="image" src="https://user-images.githubusercontent.com/68471610/229350141-e681f105-471e-4b27-8b88-a28560e4d4a6.png">

Reply:
Makes sense.